Microsoft.Alm.Authentication/Secret.cs

```diff
@@ -54,6 +54,22 @@ public static string UriToUrl(TargetUri targetUri, string @namespace)
             return targetName;
         }
 
+        /// <summary>
+        ///     Generate a key based on the ActualUri.
+        ///     This may include username, port, etc
+        /// </summary>
+        public static string UriToActualUri(TargetUri targetUri, string @namespace)
+        {
+            BaseSecureStore.ValidateTargetUri(targetUri);
+            if (String.IsNullOrWhiteSpace(@namespace))
+                throw new ArgumentNullException(@namespace);
+
+            string targetName = $"{@namespace}:{targetUri.ActualUri.AbsoluteUri}";
```

Comment:
Curiosity: why `.ActualUri` and not rely on `.ToString(false, false, false)` or similar?

It seems to me that `.ActualUri` will produce the correct results; I'm mostly just trying to understand the reasoning here.